### ISSUE TITLE
Various Rendering Issues

### ISSUE THREAD

**Issue by ChloeDawn** [asiekierka/PrestonMC#6]
**Preston version:** [0.3.3](https://minecraft.curseforge.com/projects/preston/files/2534960)
**Forge version:** [1.12.2-14.23.2.2618](http://files.minecraftforge.net/maven/net/minecraftforge/forge/1.12.2-14.23.2.2618/forge-1.12.2-14.23.2.2618-changelog.txt)

- Unreliable depth sorting results in the overlay quads being culled at odd times. 
 - Compressed blocks do not cull adjacent blocks, only themselves.
 - Due to being drawn to the translucent layer, particles render weirdly.

I think the first and last issues could be resolved by rewriting the texture png's and caching them for the models, rather than drawing translucent layers over the base texture.

**Recording of quad culling (took a few attempts to produce):**

![quad_culling](https://thumbs.gfycat.com/PartialWhimsicalArieltoucan-size_restricted.gif)

**Comment by InsomniaKitten** [comment]
You wouldn't necessarily need to stitch them into the atlas; you could get the original texture, rewrite the data, store the png data to a cache, and use it when generating the dynamic model, no?

**Comment by asiekierka** [comment]
For static rendering, it must be on the texture atlas - unless you want a TESR for every Preston block.

**Comment by InsomniaKitten** [comment]
No, it can be stored as a custom TextureAtlasSprite and it will be usable with a baked model.

**Comment by asiekierka** [comment]
Not in static rendering. How do you plan to rebind the texture? TextureAtlasSprites only store UV coordinates, not texture addresses.
